torch/csrc/jit/codegen/cuda/transform_rfactor.cpp

```diff
@@ -211,7 +211,7 @@ TensorDomain* TransformRFactor::runReplay(
   // replay marked as rfactor axes, those marked in the axes set will be
   // reduction=false
 
-  auto rfactor_root_vals = IterVisitor::getInputsTo(
```

Comment:
This is the root cause of the problem. Getting the input root domains of rfactor axes is not enough, because there can be root domains indirectly used.